### PR TITLE
Remove Test Cases from Search Result and New Search bar where they could be searched

### LINKED ISSUE #67: Remove test cases from search results
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request

**Description**:

If I search up something such as test in the search, I get results that include the testing of the algorithms in cosmos. Users don't really care about this, so such results should be filtered out.

**Expected Behavior**:

N/A

**Actual Behavior**:

You see test results

**Steps to Reproduce**:

- Search up test in the search. Some results are from the test folder of cosmos.

**Screenshots (if any)**:

**Would you like to work on this issue?**
- [ ] Yes
- [x] No

### PR DESCRIPTION
**Pull request by pragyach**
**Checklist**

- [x] My branch is up-to-date with the upstream `develop` branch.
- [x] I have added necessary documentation (if appropriate).

**Which issue does this PR fix?**: 
fixes: #67

**Why do we need this PR?**:

This pr completely solves issue#67.Test cases are removed from main search engine.Second search bar is enabled for handling these testing algorithms so they do not go waste.

**Demo (optional)**:
Before:-
cs97
After:-When test is typed in first  search bar i.e. main search bar.
fgh

When test is typed in second search bar
csf

**Testing instructions:**

No more work is left as it was discussed with @arnavb.But could work on better way to display these test cases.

**TODOs (if any)**:

### CODE REVIEW

File: search/views.py [hunk 1]
```diff
@@ -83,44 +48,103 @@ def query(request):
...
-    else:
-        return render(request, 'cosmos/searchresults.html',
-                      {'amount': len(ans),
-                       'result': ans,
-                       'recommend': rec[0:5],
-                       'query': query
-                       })
-
-
-# Search strategy
```
Comment (iattempt) [inline search/views.py]: Some codes of newer commit have overridden.

File: search/views.py [hunk 2]
```diff
@@ -1,59 +1,24 @@
...
-    return searchTag
-
-
 def index(request):
     algo_tag = searchbar()
-    algo = searchSuggestion(request)
-    if request.is_ajax():
-        mimetype = 'application/json'
-        return HttpResponse(algo, mimetype)
-    return render(request, 'cosmos/index.html', {'algo_name': algo_tag})
```
Comment (iattempt) [inline search/views.py]: Same here.

File: search/views.py [hunk 3]
```diff
@@ -1,59 +1,24 @@
...
-            results.append(tag_json)
-            i = i + 1
-            if i >= 6:
-                break
-        searchTag = json.dumps(results)
-    else:
-        searchTag = 'fail'
-    return searchTag
-
-
```
Comment (iattempt) [inline search/views.py]: And other newer support for Ajax.

**Review by soc221b — CHANGES_REQUESTED** [review]
The codes of newer commit have overridden.

File: search/views.py [hunk 4]
```diff
@@ -83,44 +48,103 @@ def query(request):
...
+    return render(request, 'cosmos/searchresults.html',
+                  {'amount': len(ans), 'result': ans, 'recommend': rec[0:5], 'query': query})
+
+def searchbar1():
+    jsonFile = open('tags.json', 'r')
+    data = json.load(jsonFile)
+    algo_list = data['tags']
+    r_no = random.randint(0,len(algo_list))
+    algo_tag1 = algo_list[r_no]
+    return algo_tag1
```
Comment (iattempt) [inline search/views.py]: What is different between `searchbar()` and `searchbar1()`?

File: search/views.py [hunk 5]
```diff
@@ -83,44 +48,103 @@ def query(request):
...
+def error403(request):
+    return render(request, 'cosmos/error/HTTP403.html')
+
+
+def error404(request):
+    return render(request, 'cosmos/error/HTTP404.html')
+
+
+def error500(request):
+    return render(request, 'cosmos/error/HTTP500.html')
```
Comment (iattempt) [inline search/views.py]: These function is duplicate.

File: search/views.py [hunk 6]
```diff
@@ -83,44 +48,103 @@ def query(request):
...
+
+
+# Search query
+def query1(request):
```
Comment (iattempt) [inline search/views.py]: Should be readable and meaningful.

**Review by soc221b — CHANGES_REQUESTED** [review]

**Comment by AdiChat** [comment]
Nice work :+1: 

Having two search bars will confuse users.

Ideally, users may not search for test files and might want to go to particular implementations. That being said, we can associate test files with respective codes and display them when the codes are displayed within Cosmos Search (a future feature). 
Associating test files will give users additional insights into the various possible test case scenarios for the particular algorithm.

**Pull request closed without merging**